schematic/manifest/generator.py

```diff
@@ -698,7 +695,7 @@ def get_manifest(self, dataset_id: str = None, sheet_url: bool = None, json_sche
             if syn_id_and_path and sheet_url:
 
                 # get synapse ID manifest associated with dataset
-                manifest_data = syn.get(syn_id_and_path[0], downloadLocation=manifests_folder_path, ifcollision="overwrite.local")
+                manifest_data = syn_store.syn.get(syn_id_and_path[0], downloadLocation=manifests_folder_path, ifcollision="overwrite.local")
```

Comment:
Let's try minimizing synapse references here and work only with SynapseStorage methods; one approach
- add a boolean-flag argument to SynapseStorage's getDatasetManifest that indicates whether to fetch the manifest data or not
- if flag true, return the manifest data (e.g. instead of the synid) using the call below  
syn_store.syn.get(syn_id_and_path[0], downloadLocation=manifests_folder_path, ifcollision="overwrite.local")
--- by default the flag can be false (for backward compatibility)

That way we'll have all synapse logic encapsulated in SynapseStorage and here we'll only focus on the logic relevant for the manifest population; further that would avoid the repeating lines 698 and 710. You can still check if the manifest data is returned properly by getDatasetManifest similarly to what you do in l695 and l709.

Reply:
This sounds like a good idea. I was trying to think of a [`Decorators`](https://book.pythontips.com/en/latest/decorators.html)-based approach to it. To create a decorator (maybe an `@get_manifest_data` decorator) or something, but that would be useful only to this function, not sure if any other methods could/would be decorated this way.

I've implemented the it per your recommendation above @milen-sage, which makes more sense.